### PR TITLE
EVG-7234 remove legacy settings field

### PR DESCRIPTION
**Pull request by ablack12**
This is just the last ticket in the epic, meant to remove the risk of maintaining two similar but different fields for a distro, where ProviderSettings will be just one document from ProviderSettingsList. 

I'll make sure all distros have providerSettingsList populated, and then in staging will remove the legacy providerSettings to ensure everything still works correctly, and then will remove the legacy in prod before merging this PR.

### CODE REVIEW

**Comment by ablack12** [comment]
Note on the aggregation change: can't compare hosts directly because for some reason the actual pointers for ProviderSettingsList are different but the values are the same so that seems like enough?

File: cloud/ec2_integration_test.go [hunk 1]
```diff
@@ -6,6 +6,8 @@ import (
 	"context"
 	"testing"
 
+	"github.com/evergreen-ci/birch"
+
```
Comment (tychoish) [inline cloud/ec2_integration_test.go:10]: nit: blank line.

File: model/host/host_test.go [hunk 2]
```diff
@@ -4568,37 +4580,45 @@ func TestStartingHostsByClient(t *testing.T) {
...
 			assert.Fail(t, "unrecognized client options")
 		}
 	}
 }
 
+func compareHosts(t *testing.T, host1, host2 Host) {
+	assert.Equal(t, host1.Id, host2.Id)
+	assert.Equal(t, host1.Status, host2.Status)
+	assert.Equal(t, host1.Distro.Provider, host2.Distro.Provider)
+	assert.Equal(t, host1.Distro.ProviderSettingsList[0].ExportMap(), host2.Distro.ProviderSettingsList[0].ExportMap())
```
Comment (tychoish) [inline model/host/host_test.go:4619]: does this really work? 

I think there's an "Equal" method on birch documents that might be useful here,

Comment (ablack12) [reply model/host/host_test.go:4619]: There's an Equal method on Elements and Values but not on documents right now, and comparing documents with just the assert Equals complains about the pointers being different. This does work though!

**Pull request merged**